tests/utils.py

```diff
@@ -40,7 +40,7 @@ def patch_cwd(set_cwd: Path) -> Generator[None, None]:
 
 
 @contextlib.contextmanager
-def patch_execution(*args: str, set_cwd: Optional[Path] = None) -> Generator[None, None]:
+def patch_execution(*args: str, set_cwd: Path | None = None) -> Generator[None, None]:
```

Comment:
[PEP 604](https://peps.python.org/pep-0604/) for union types using `|` wasn't added until Python 3.10 but prospector still [supports 3.9](https://github.com/prospector-dev/prospector/blob/master/pyproject.toml#L26) so I suggest reverting this line unless dropping 3.9 is the plan. Since 3.9 is not quite end of life yet, I'd like to avoid that.

Reply:
I'm on mobile si it's hard to check but there might be an import from the future (form future import annotations)

Reply:
@Pierre-Sassoulas You're right, there is, I missed that since it was collapsed by the GitHub diff UI :-)

Reply:
Then finally it's OK :-)